### PR TITLE
Support for password-protected games

### PR DESCRIPTION
**Pull request by AlexNisnevich**
image

### CODE REVIEW

File: src/common/components/play/PreGameModal.js [hunk 1]
```diff
@@ -89,14 +93,26 @@ export default class PreGameModal extends React.Component {
...
+    this.setState({ enteredPassword: e.target.value });
+  }
+
   handleStartGame = () => {
-    this.props.onStartGame(this.format.name, this.deckForGame);
+    const { options, onStartGame } = this.props;
+    const { enteredPassword } = this.state;
+
+    if (options.passwordToJoin && enteredPassword !== options.passwordToJoin) {
+      alert('Incorrect password!');
```
Comment (jacobnisnevich) [inline src/common/components/play/PreGameModal.js]: Should this be displayed in the modal rather than as an alert?

Comment (AlexNisnevich) [reply src/common/components/play/PreGameModal.js]: I can't think of a good way to do that that's not gross :-/

File: src/common/components/play/multiplayer/GameRow.js [hunk 2]
```diff
@@ -35,8 +36,9 @@ export default class GameRow extends React.Component {
   }
 
   handleJoinGame = () => {
-    const { game: { id, name, format }, onJoinGame } = this.props;
-    onJoinGame(id, name, GameFormat.fromString(format));
+    const { game: { id, name, format, options }, onJoinGame } = this.props;
```
Comment (jacobnisnevich) [inline src/common/components/play/multiplayer/GameRow.js]: Woah you can do that??

Comment (AlexNisnevich) [reply src/common/components/play/multiplayer/GameRow.js]: 😁

Comment (AlexNisnevich) [reply src/common/components/play/multiplayer/GameRow.js]: The more you know ...

**Pull request merged**